src/scanners/javascript.js

```diff
@@ -30,54 +30,70 @@ export default class JavaScriptScanner {
     _messages=messages,
   }={}) {
     return new Promise((resolve) => {
-      // ESLint is synchronous and doesn't accept streams, so we need to
-      // pass it the entire source file as a string.
-      var eslint = _ESLint.linter;
-
-      for (const name in _rules) {
-        this._rulesProcessed++;
-        eslint.defineRule(name, _rules[name]);
-      }
-
-      var report = eslint.verify(this.code, {
+      var cli = new _ESLint.CLIEngine({
         env: { es6: true },
-        parserOptions: { ecmaVersion: 2017 },
+        baseConfig: {
+          parserOptions: { ecmaVersion: 2017 },
+          settings: {
+            addonMetadata: this.options.addonMetadata,
+          },
+        },
         ignore: false,
         rules: _ruleMapping,
-        settings: {
-          addonMetadata: this.options.addonMetadata,
-        },
-      }, {
+        plugins: ['no-unsafe-innerhtml'],
         allowInlineConfig: false,
         filename: this.filename,
+        // Avoid loading the addons-linter .eslintrc file
+        useEslintrc: false,
```

Comment:
Presumably this change also means that eslintrc's nested in a webext can't override anything. It would be good to add an explicit test for something like that.